[15.0][FIX] website_sale_hide_price: fix key error when editing snippet

pilarvargas-tecnativa opened the pull request:
When editing the website, a problem was encountered when inserting and editing the dynamic product snippet. When changing the filter to "product accessories" or "recently sold products", an error occurred accessing the "website_hide_price" key if it didn't exist. The solution involves modifying the access to the "website_hide_price" key, thus avoiding the mentioned error. With this correction, the proper functioning of dynamic product snippets is guaranteed.

![image](https://github.com/OCA/e-commerce/assets/118818446/9eeac2fa-82f7-4d4e-b97c-7c577d00cbb0)

cc @Tecnativa TT43493

@chienandalu @CarlosRoca13 please review :)

pilarvargas-tecnativa committed:
[FIX] website_sale_hide_price: fix key error when editing snippet

When editing the website, a problem was encountered when inserting and
editing the dynamic product snippet. When changing the filter to
"product accessories" or "recently sold products", an error occurred
accessing the "website_hide_price" key if it didn't exist. The solution
involves modifying the access to the "website_hide_price" key, thus
avoiding the mentioned error. With this correction, the proper
functioning of dynamic product snippets is guaranteed.

TT43493

minhthie commented:
Functional test: lgtm

chienandalu reviewed (commented):
To improve the coverage of the widget, we should also hide the add to cart button in the different templates available:

https://github.com/OCA/OCB/blob/15.0/addons/website_sale/data/product_snippet_template_data.xml#L32

Discounted price should be hidden as well:

https://github.com/odoo/odoo/blob/15.0/addons/website_sale/data/product_snippet_template_data.xml#L288-L289

pilarvargas-tecnativa committed:
[IMP] website_sale_hide_price: Hide add to cart button and prices

Hide add to cart button and prices from dynamic product snippet
templates if not website_show_price

When inheriting templates from the snippet, they were duplicated in
the web editor selector. To avoid this, they have been defined in "data"
like odoo does and the id's of the child templates have been modified.

TT43493

pedrobaeza commented:
/ocabot merge minor

chienandalu commented:
Please fw port to v16
